### DISCUSSION TITLE
Bulk undefine functions

### DISCUSSION THREAD

**Discussion by windnwavefan** [Vector35/binaryninja-api#4591] [General] [answered]
Not sure if this belongs in the API discussions but I have an ELF file where binja has incorrectly created a bunch functions based on the presence of null bytes. Is it possible to bulk undefine functions?

I should also mention the architecture is aarch64 and it's interpreting the null bytes as `udf     #0`

Also i have assigned a hotkey but my scroll wheel is starting to get worn out

**Comment by lwerdna** [comment, marked as answer]
I don't know a UI way to bulk undefine function, you'd probably have to script it. But as of https://github.com/Vector35/arch-arm64/commit/f40a735246e8a3100a5caaeed823553ddd41e917, stray nulls should not be so easily considered instructions which cause functions to be made. Here's a before on the left, and after on the right:

![image](https://github.com/Vector35/binaryninja-api/assets/474191/218d3c40-35e9-4ebf-a111-422f20b164b1)

Update should be on dev soon.

**Comment by psifertex** [comment]
As to the batch-remove, probably something like something like:

```
for f in bv.functions:
  if bv.read(f.start, 4) == b'\0\0\0\0':
    bv.remove_function(f)
```
